backend/ibutsu_server/controllers/login_controller.py

```diff
@@ -0,0 +1,263 @@
+        return "Bad request", 400
+    code = connexion.request.args["code"]
+    frontend_url = build_url(
+        current_app.config.get("FRONTEND_URL", "http://localhost:3000"), "login"
+    )
+    if provider == "keycloak":
+        # Do the kc stuff
+        provider_config = get_keycloak_config(is_private=True)
+    else:
+        provider_config = get_provider_config(provider, is_private=True)
+    user = None
+    if provider == "google":
+        # Google does things its own way...
+        try:
+            # Verify the token
+            id_info = id_token.verify_oauth2_token(code, Request(), provider_config["client_id"])
+            # ID token is valid. Get the user's Google Account ID from the decoded token.
+            user = get_user_from_provider(provider, id_info)
+        except ValueError:
+            # Invalid token
+            return "Unauthorized", 401
+    else:
+        # For everyone else
+        payload = {
+            "client_id": provider_config["client_id"],
+            "code": code,
+            "grant_type": "authorization_code",
+            "redirect_uri": provider_config["redirect_uri"],
+        }
+        if provider_config.get("client_secret"):
+            payload["client_secret"] = provider_config["client_secret"]
+        response = requests.post(
+            provider_config["token_url"], data=payload, headers={"Accept": "application/json"}
+        )
+        if response.status_code == 200:
+            if provider == "keycloak":
+                user = get_user_from_keycloak(response.json())
+            else:
+                user = get_user_from_provider(provider, response.json())
+    if not user:
+        return "Unauthorized", 401
+    jwt_token = generate_token(user.id)
+    token = Token.query.filter(Token.name == "login-token", Token.user_id == user.id).first()
+    if not token:
+        token = Token(name="login-token", user_id=user.id)
+    token.token = jwt_token
+    session.add(token)
+    session.commit()
+    if provider == "keycloak":
+        query_params = urlencode({"email": user.email, "name": user.name, "token": jwt_token})
+        return redirect(f"{frontend_url}?{query_params}")
+    elif provider == "google":
+        return {"email": user.email, "name": user.name, "token": jwt_token}
+    else:
+        return make_response(
+            AUTH_WINDOW.format(
+                data=json.dumps({"email": user.email, "name": user.name, "token": jwt_token})
+            )
+        )
+    # Finally, if nothing above works or is caught, return a 401
+    return "Unauthorized", 401
+
+
+def register(email=None, password=None):
+    """Register a user
+
+    :param email: The e-mail address of the user
+    :type email: str
+    :param password: The password for the user
+    :type password: str
+    """
+
+    # Create a random activation code. Base64 just for funsies
+    activation_code = urlsafe_b64encode(bytes(uuid4())).strip(b"=")
```

Comment:
This is where the registration error is occuring:
```
  File "/home/jdupuy/iqe/iqe-repos/ibutsu/ibutsu-server/backend/.ibutsu-env/lib/python3.9/site-packages/connexion/decorators/uri_parsing.py", line 149, in wrapper
    response = function(request)
  File "/home/jdupuy/iqe/iqe-repos/ibutsu/ibutsu-server/backend/.ibutsu-env/lib/python3.9/site-packages/connexion/decorators/validation.py", line 193, in wrapper
    response = function(request)
  File "/home/jdupuy/iqe/iqe-repos/ibutsu/ibutsu-server/backend/.ibutsu-env/lib/python3.9/site-packages/connexion/decorators/parameter.py", line 115, in wrapper
    return function(**kwargs)
  File "/home/jdupuy/iqe/iqe-repos/ibutsu/ibutsu-server/backend/ibutsu_server/controllers/login_controller.py", line 176, in register
    activation_code = urlsafe_b64encode(bytes(uuid4())).strip(b"=")
```
Steps to reproduce:
1) Navigate to http://localhost:3000/login
2) Click "sign up"
3) Enter email and password
4) Hit "Register"

At step 4) I see "Failed to fetch" in the FE and the backend shows the above traceback.

Reply:
Ah, again I was looking in the wrong place -_-